src/main/java/com/bootcamp/demo/controller/CardController.java

```diff
@@ -0,0 +1,32 @@
+package com.bootcamp.demo.controller;
+
+import com.bootcamp.demo.service.CardService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Map;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RestController
+@RequestMapping(value = "/card", produces = APPLICATION_JSON_VALUE )
+public class CardController {
+    private final CardService cardService;
+
+    public CardController(CardService cardService) {
+        this.cardService = cardService;
+    }
+
+    @DeleteMapping(path="{id}")
+    ResponseEntity<String> removeCard(@PathVariable String id){
+      this.cardService.removeCard(id);
+      return new ResponseEntity<>("Successfully deleted card "+ id, HttpStatus.OK);
+    }
+
+    @PutMapping(path="{id}", consumes = APPLICATION_JSON_VALUE)
+    ResponseEntity<String> updateCard(@PathVariable String id, @RequestBody Map cardDetails){
```

Comment:
The `cardDetails` parameter should actually be of type `Card`?
The controller methods could be made public.

Reply:
I made the card details of type Map because I made the update method in FirebaseDao reusable for multiple collections, giving it the id, collectionName and Map for new data. I could make it more specific to work only for cards if needed. I'll also make the methods public. The remove method in the dao is also reusable for multiple collections as is.

Reply:
Good ideea! The DAO can be made generic/reusable and it's a good ideea for a DAO to accept/handle "primitives" but all the layers up to the DAO (Controller, Service, Repository) should work with model objects.
Usually the Repository handles objects persistence and can work with DAOs to achieve this low level data access. For example all repositories from Spring Data are generics that can handle "model" objects: https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/Repository.html